arch/src/main/java/knaufdan/android/arch/databinding/recyclerview/ComponentAdapter.kt

```diff
@@ -0,0 +1,54 @@
+package knaufdan.android.arch.databinding.recyclerview
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.RecyclerView
+import knaufdan.android.arch.base.component.IComponent
+import knaufdan.android.arch.utils.findLifecycleOwner
+
+class ComponentAdapter<DataSource>(
+    components: List<IComponent<DataSource>>
+) : RecyclerView.Adapter<BindableViewHolder<DataSource>>() {
+    // Store data in separate list to lose the reference and prevent error if references changes.
+    private val dataSource = components.toList()
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): BindableViewHolder<DataSource> =
+        dataSource[viewType].run {
+            val binding = DataBindingUtil.inflate<ViewDataBinding>(
+                LayoutInflater.from(parent.context),
+                getLayoutRes(),
+                parent,
+                false
+            ).apply {
+                parent.context.findLifecycleOwner()?.apply {
+                    lifecycleOwner = this
+                }
+            }
```

Comment:
Create `abstract class` as base for `ComponentAdapter` and `BindableAdapter`

Needed methods:
- createBinding()
- getBindingKey()
- getDataValue(position:Int)

General handling for `onViewRecycled()` in `BindableViewHolder`

Reply:
done.